### DISCUSSION TITLE
Why is my -gravity setting being ignored in this command?

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4612] [Help] [answered]
Why is my -gravity setting for the text (-size 550x _**-gravity center**_ caption:) being ignored in this command?

```
for %F in (*.jpg) do magick "%F" ^
-resize 572x ^
+write mpr:img +delete ^
( mpr:img -blur 10x10 -resize x592 -gravity center -crop 592x592+0+0 +repage ) ^
( mpr:img -bordercolor #DB9D00 -border 1x1 -bordercolor #FDF9DC -border 1x1 -bordercolor #DB9D00 -border 1x1 +repage ) ^
-gravity center -compose over -composite ^
-bordercolor #DB9D00 -border 2x2 -bordercolor #FDF9DC -border 2x2 -bordercolor #DB9D00 -border 2x2 ^
-background none -fill blue -font San-Diego -pointsize 70 -size 550x -gravity center caption:"MARLINGBAITS" -flatten ^
"result_%F"
```

Result:
![result_maxresdefault (1)](https://user-images.githubusercontent.com/95970582/146668143-fcf65a5c-3b33-4c98-a0ed-99b5a45ec820.jpg)

As you can see the text is set in the NorthWest not Center. (IM7Win10)

**Comment by snibgo** [comment, marked as answer]
"-flatten" is not sensitive to "-gravity". However, "-composite" is.

**Reply by SentientBeingOnAQuest** [reply]
Changing -flatten to -composite has fixed the issue. Thank you!
